12_4.cpp

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include <cmath>
+#include <fstream>
+using namespace std;
+
+
+int main() {
+    int n;
+    cout << "n = ";
+    cin >> n;
+    int* int_array = (int*)malloc(sizeof(int)*n);
+    for (int i=0; i<n; i++) {
+        cout << "m(" << i << ") = ";
+        cin >> int_array[i];
+    }
+
+    double* double_array = (double*)malloc(sizeof(double)*n);
+    ifstream fin("12_4_in.txt");
+    if (!fin) {
+        cout << "File hasn't opened!";
+        free(int_array);
+        free(double_array);
+        fin.close();
+        return -1;
+    }
+    double item;
+    for (int i=0; i<n; i++) {
+        if (fin.eof()) {
+            cout << "Not enough items in file!";
+            free(int_array);
+            free(double_array);

```

Comment:
looks like it is possible memory leak (delete inside loop and if block is dangerous)

Reply:
but here it is ok